code/ui/richpanel/RichPanel.Resizeable.cs

```diff
@@ -0,0 +1,331 @@
+using System;
+
+using Sandbox;
+using Sandbox.UI;
+
+namespace TTTReborn.UI
+{
+    public struct BoxData
+    {
+        public float Width;
+        public float Height;
+        public float Top;
+        public float Right;
+        public float Bottom;
+        public float Left;
+        public Vector2? TransformRatio;
+    }
+
+    public partial class RichPanel : TTTPanel
+    {
+        public bool IsDraggable { get; set; } = false;
+
+        public enum DragAnchor
+        {
+            TOPLEFT,
+            TOP,
+            TOPRIGHT,
+            RIGHT,
+            BOTTOMRIGHT,
+            BOTTOM,
+            BOTTOMLEFT,
+            LEFT
+        }
+
+        public bool IsDragging { get; private set; } = false;
+
+        public bool CanStartDragging
+        {
+            get
+            {
+                return _canStartDragging && IsDraggable;
+            }
+            private set
+            {
+                _canStartDragging = value;
+
+                SetClass("draggable", CanStartDragging);
+            }
+        }
+        private bool _canStartDragging = false;
+
+        public DragAnchor? CurrentDragAnchor
+        {
+            get
+            {
+                return _currentDragAnchor;
+            }
+            private set
+            {
+                _currentDragAnchor = value;
+
+                CanStartDragging = _currentDragAnchor != null;
+            }
+        }
+        private DragAnchor? _currentDragAnchor = null;
+
+        public DragAnchor? CurrentHorizontalDragAnchor
+        {
+            get
+            {
+                return _currentHorizontalDragAnchor;
+            }
+            private set
+            {
+                _currentHorizontalDragAnchor = value;
+
+                if (value != null)
+                {
+                    if (CurrentDragAnchor == DragAnchor.TOP)
+                    {
+                        CurrentDragAnchor = value == DragAnchor.LEFT ? DragAnchor.TOPLEFT : DragAnchor.TOPRIGHT;
+                    }
+                    else if (CurrentDragAnchor == DragAnchor.BOTTOM)
+                    {
+                        CurrentDragAnchor = value == DragAnchor.LEFT ? DragAnchor.BOTTOMLEFT : DragAnchor.BOTTOMRIGHT;
+                    }
+                    else
+                    {
+                        CurrentDragAnchor = value;
+                    }
+                }
+                else
+                {
+                    CurrentDragAnchor = CurrentVerticalDragAnchor;
+                }
```

Comment:
```suggestion
                _currentHorizontalDragAnchor = value ?? CurrentVerticalDragAnchor;

                    if (CurrentDragAnchor == DragAnchor.TOP)
                    {
                        CurrentDragAnchor = value == DragAnchor.LEFT ? DragAnchor.TOPLEFT : DragAnchor.TOPRIGHT;
                    }
                    else if (CurrentDragAnchor == DragAnchor.BOTTOM)
                    {
                        CurrentDragAnchor = value == DragAnchor.LEFT ? DragAnchor.BOTTOMLEFT : DragAnchor.BOTTOMRIGHT;
                    }
                    else
                    {
                        CurrentDragAnchor = value;
                    }
```

Can this work, haven't taken an in-depth look.

Reply:
nvm sec, let me throw up a better snippet.

Reply:
Please test these changes :) If I'm looking into it, I think last line in the last else should be `CurrentDragAnchor 
 = _currentHorizontalDragAnchor;`

Reply:
```csharp
        public DragAnchor? CurrentHorizontalDragAnchor
        {
            get
            {
                return _currentHorizontalDragAnchor;
            }
            private set
            {
                _currentHorizontalDragAnchor = value ?? CurrentVerticalDragAnchor;

                switch (CurrentDragAnchor)
                {
                    case DragAnchor.TOP:
                        CurrentDragAnchor = value == DragAnchor.LEFT ? DragAnchor.TOPLEFT : DragAnchor.TOPRIGHT;
                        break;
                    case DragAnchor.BOTTOM:
                        CurrentDragAnchor = value == DragAnchor.LEFT ? DragAnchor.BOTTOMLEFT : DragAnchor.BOTTOMRIGHT;
                        break;
                    default:
                        CurrentDragAnchor = _currentHorizontalDragAnchor;
                        break;
                }
            }
        }
        private DragAnchor? _currentHorizontalDragAnchor = null;
```

Reply:
But isn't the switch running even if `value` is null? So what if `CurrentDragAnchor` is `DragAnchor.TOP`? Then we would get `DragAnchor.TOPRIGHT` i think.